lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -131,6 +169,37 @@ def index(
             trans, serialization_params, filter_query_params, deleted_only=deleted, all_histories=all
         )
 
+    @router.get(
+        "/api/histories/query",
```

Comment:
Ehm, that should've just been `/api/histories` ...

Reply:
We could do that, but that would require changing/deprecating an existing endpoint.

Reply:
How so ? You only need to add the additional filters, this is really not worth a separate route.